packages/lesswrong/server/rateLimitUtils.ts

```diff
@@ -216,3 +232,75 @@ export async function getRecentKarmaInfo (userId: string): Promise<RecentKarmaIn
   const allVotes = await votesRepo.getVotesOnRecentContent(userId)
   return calculateRecentKarmaInfo(userId, allVotes)
 }
+
+async function getVotesForComparison(userId: string, currentVotes: RecentVoteInfo[]) {
+  currentVotes = currentVotes.sort((a, b) => moment(b.votedAt).diff(a.votedAt));
+  const [mostRecentVoteInfo] = currentVotes;
+
+  const comparisonVotes = [...currentVotes];
+  comparisonVotes.shift();
+
+  if (documentOnlyHasSelfVote(userId, mostRecentVoteInfo, currentVotes)) {
+    // Check whether it was a self-vote on a post or comment
+    const { collectionName } = mostRecentVoteInfo;
+    // Fetch all the votes on the post or comment that would've been pushed out of the 20-item window by this one, and use those instead
+    const votesRepo = new VotesRepo();
+    const votesOnNextMostRecentDocument = await votesRepo.getVotesOnPreviousContentItem(userId, collectionName, mostRecentVoteInfo.postedAt);
+    comparisonVotes.push(...votesOnNextMostRecentDocument);
+  }
+
+  return comparisonVotes;
+}
+
+function triggerReviewForStricterRateLimits(
+  userId: string,
+  commentRateLimitComparison: RateLimitComparison<CommentAutoRateLimit>,
+  postRateLimitComparison: RateLimitComparison<PostAutoRateLimit>,
+  context: ResolverContext
+) {
+  if (!commentRateLimitComparison.isStricter && !postRateLimitComparison.isStricter) {
+    return;
+  }
+
+  if (commentRateLimitComparison.isStricter) {
+    const { strictestNewRateLimit: { itemsPerTimeframe, timeframeUnit, timeframeLength } } = commentRateLimitComparison;
+
+    void triggerReview(userId);
+    void appendToSunshineNotes({
+      moderatedUserId: userId,
+      adminName: 'Automod',
+      text: `User triggered a stricter ${itemsPerTimeframe} comment(s) per ${timeframeLength} ${timeframeUnit} rate limit`,
+      context,
+    });
+  }
+
+  if (postRateLimitComparison.isStricter) {
+    const { strictestNewRateLimit: { itemsPerTimeframe, timeframeUnit, timeframeLength } } = postRateLimitComparison;
+
+    void triggerReview(userId);
+    void appendToSunshineNotes({
+      moderatedUserId: userId,
+      adminName: 'Automod',
+      text: `User triggered a stricter ${itemsPerTimeframe} post(s) per ${timeframeLength} ${timeframeUnit} rate limit`,
+      context,
+    });
+  }
+}
+
+export async function checkForStricterRateLimits(userIds: string[], context: ResolverContext) {
+  // We can't use a loader here because we need the user's karma which was just updated by this vote
+  const votedOnUsers = await Users.find({ _id: { $in: userIds } }).fetch();
+
+  const votesRepo = new VotesRepo();
+
+  void Promise.all(votedOnUsers.map(async (user) => {
+    const allVotes = await votesRepo.getVotesOnRecentContent(user._id);
+    console.log({ allVotes });
```

Comment:
lint error is real

Reply:
Yeah, sorry, was just debugging the integration tests (which, turns out, actually caught real issues!)